DNSSEC RR's support in Bro

fatemabw opened the pull request:
Just implemented DNSSEC RRs type support- DNSSEC RR types NSEC, DNSKEY and DS.
Feedback requested!
Thanks!
Fatema.

0xxon commented on src/analyzer/protocol/dns/DNS.cc:
```diff
@@ -182,7 +182,7 @@ int DNS_Interpreter::ParseQuestion(DNS_MsgInfo* msg,
 		}
 
-	EventHandlerPtr dns_event = nullptr;
+	EventHandlerPtr dns_event = 0;
```
Is there a reason for the change from nullptr to 0? nullptr is more modern...

0xxon commented:
Hi Fatema,

this actually looks pretty good to me. Would you mind perhaps adding btests that test the new behavior? :)

fatemabw commented:
Actually, I think that's what set in Bro 2.5.4 src code (as I pulled down that src n not the master) , as I didn't make that change but noticed that while requesting a merge :) 

So no reason why can't it be set to what the master branch already have :) sorry for the confusion, will revert it back to its original value...

Thanks,
Fatema 


Sent from my iPhone 5

> On Aug 28, 2018, at 6:02 PM, Johanna <notifications@github.com> wrote:
> 
> @0xxon commented on this pull request.
> 
> In src/analyzer/protocol/dns/DNS.cc:
> 
> > @@ -182,7 +182,7 @@ int DNS_Interpreter::ParseQuestion(DNS_MsgInfo* msg,
>  		return 0;
>  		}
>  
> -	EventHandlerPtr dns_event = nullptr;
> +	EventHandlerPtr dns_event = 0;
> Is there a reason for the change from nullptr to 0? nullptr is more modern...
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

fatemabw commented:
Thanks Johanna for taking a look at it :) 
I was reading about creating btest test cases on Bro documentation, didn't notice that before, will add some tests for the new code ...

Thanks,
Fatema 

Sent from my iPhone 5

> On Aug 28, 2018, at 6:03 PM, Johanna <notifications@github.com> wrote:
> 
> Hi Fatema,
> 
> this actually looks pretty good to me. Would you mind perhaps adding btests that test the new behavior? :)
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

0xxon commented:
Also - if you did not do it already - you should probably let the test-suite that already exists run. CI shows some crashes in DNS tests with your changes :).

If you have any problems getting that to run, or problems writing btests, let me know.

fatemabw commented:
Thanks Johanna for helping troubleshooting the issues :-)
Got rid of a small error in btest:
 % 'bro -r $TRACES/dns-two-responses.trace' failed unexpectedly (exit code -6)
  % cat .stderr
  src/tcmalloc.cc:284] Attempt to free invalid pointer 0x9053b0

caused because of naming conflict of "signature" struct variable and "signature" local variable of type BroString inside the ParseRR_RRSIG function in DNS.cc. Updated DNS.cc. 

The btest test came out clean this time, with the scripts.base.protocols.dns.duplicate-reponses and  scripts.base.protocols.dns.dns-key baseline update on local test system.
 
Thanks,
Fatema

fatemabw commented:
Updated the code for DNSSEC RR's parsing support in Bro. 
Also, uploaded a new pcap for dnssec testing, which has DS, DNSKEY and RRSIG requests and response, which can be tested with new code. And added corresponding btest script to test the pcap in scripts/base/protocols/dns/dnssec.bro

Thanks,
Fatema.

jsiwek commented:
A meta-comment: is there a reason for having so many commits?  Probably this type of patch just warrants one, not 40+.  Also, if you're still working on it, I'd suggest just closing the PR and then creating a new one once you have a finalized version ready to consider.  We're usually not that strict about it, but in this case I'd really suggest squashing the number of commits down at that time.

fatemabw commented:
Hey Jon,

Makes sense, sorry for the changes and so many commits, pretty new to
github and wasn't sure how things work on github land ( learning slowly) :)
I am almost done, just added two new btests with files, will go ahead and
close the PR and request a new pull request. :)

Thanks for the suggestions!

Fatema.

On Wed, Sep 5, 2018 at 11:51 AM Jon Siwek <notifications@github.com> wrote:

> A meta-comment: is there a reason for having so many commits? Probably
> this type of patch just warrants one, not 40+. Also, if you're still
> working on it, I'd suggest just closing the PR and then creating a new one
> once you have a finalized version ready to consider. We're usually not that
> strict about it, but in this case I'd really suggest squashing the number
> of commits down at that time.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bro/bro/pull/146#issuecomment-418782076>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AGmkHW5Mv1vmGgt8ZDwT9dDSDWRqxUh-ks5uX_L9gaJpZM4WObi7>
> .
>

fatemabw commented:
Closing this PR as per Jon's suggestion :)
